be/src/olap/tablet_meta.cpp

```diff
@@ -544,6 +548,27 @@ OLAPStatus TabletMeta::add_inc_rs_meta(const RowsetMetaSharedPtr& rs_meta) {
     return OLAP_SUCCESS;
 }
 
+void TabletMeta::delete_expired_snapshot_rs_meta_by_version(const Version& version) {
+    auto it = _expired_snapshot_rs_metas.begin();
+    while (it != _expired_snapshot_rs_metas.end()) {
+        if ((*it)->version() == version) {
+            _expired_snapshot_rs_metas.erase(it);
+            break;
```

Comment:
Are you sure there will be only one version which equals to the specified version?